components/brave_news/browser/resources/shared/useFeedV2.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import usePromise from "$web-common/usePromise";
+import { useEffect, useState } from "react";
+import getBraveNewsController, { FeedV2, FeedV2Type } from "./api";
+
+export type FeedView = 'all' | `publishers/${string}` | `channels/${string}`
+
+const feedTypeToFeedView = (type: FeedV2Type | undefined): FeedView => {
+  if (type?.channel) return `channels/${type.channel.channel}`
+  if (type?.publisher) return `publishers/${type.publisher.publisherId}`
+  return 'all'
+}
+
+const FEED_KEY = 'feedV2'
+const FEED_VIEW_KEY = 'feedV2-view'
+
+const saveFeed = (feed?: FeedV2) => {
+  if (!feed) return
+
+  // Note: We have to provide a replacer, because BigInt can't be serialized to JSON
+  sessionStorage.setItem(FEED_KEY, JSON.stringify(feed, (_, value) => typeof value === "bigint"
+    ? value.toString()
+    : value))
+}
+
+const maybeLoadFeed = (view: FeedView) => {
+  const data = sessionStorage.getItem(FEED_KEY)
+  if (!data) return
+
+  const feed: FeedV2 = JSON.parse(data)
+
+  // If the feed doesn't match what we stored, don't return it.
+  return feedTypeToFeedView(feed.type) === view
+    ? feed
+    : undefined
+}
+
+export const useFeedV2 = () => {
```

Comment:
Hey, so in my head there was a bunch of related functionality for `FeedV2` and it felt natural to pull it all into one file, rather than dumping a bunch more stuff into `Context.tsx`

No strong feelings on this though, happy to move it into `Context.tsx` if you prefer that (or name it something else) :smile:

I think of this as "The info we need to be able to display the feed"